tests/Unit/controllers/account.test.js

```diff
@@ -0,0 +1,494 @@
+      };
+
+      const req = {
+        body: {
+          ...expectedUpdates,
+        },
+        auth: {},
+      };
+      const res = {
+        send: jest.fn(),
+      };
+      const next = jest.fn();
+
+      accountRepository.update = jest.fn((opts) => {
+        expect(opts.updates).toEqual(expectedUpdates);
+      });
+
+      await controller.update(req, res, next);
+
+      expect(accountRepository.update).toHaveBeenCalledTimes(1);
+      expect(res.send).toHaveBeenCalledTimes(1);
+      expect(next).toHaveBeenCalledTimes(0);
+    });
+    test("Should use user id from jwt", async () => {
+      const expectedId = "Should be called with this";
+
+      const req = { body: {}, auth: { id: expectedId } };
+      const res = {
+        send: () => {},
+      };
+
+      const next = () => {};
+
+      accountRepository.update = jest.fn((opts) => {
+        expect(opts.query).toEqual({ _id: expectedId });
+      });
+
+      await controller.update(req, res, next);
+    });
+    test("Should throw invalid update error when update contains invalid fields and handle it", async () => {
+      const expectedError = new Error(errors.INVALID_UPDATES);
+      expectedError.name = errors.INVALID_UPDATES;
+
+      const req = { body: {} };
+      const res = {};
+      const next = jest.fn();
+
+      await controller.update(req, res, next);
+
+      expect(next).toHaveBeenCalledTimes(1);
+      expect(next).toBeCalledWith(expectedError);
+    });
+    test("Should handle errors", async () => {
+      const expectedError = new Error(errors.UNKNOWN);
+      expectedError.name = errors.UNKNOWN;
+
+      accountRepository.update = jest.fn((opts) => {
+        throw expectedError;
+      });
+
+      const req = { body: {}, auth: {} };
+      const res = {};
+      const next = jest.fn();
+
+      await controller.update(req, res, next);
+
+      expect(next).toHaveBeenCalledTimes(1);
+      expect(next).toBeCalledWith(expectedError);
+    });
+  });
+
+  describe("Delete account", () => {
+    test("Should delete user account and associated profile, relationships, and server list", async () => {
+      const req = { body: {}, auth: {} };
+      const res = {
+        status: jest.fn(() => res),
+        send: jest.fn(),
+      };
+      const next = jest.fn();
+
+      accountRepository.delete = jest.fn();
+      profileRepository.delete = jest.fn();
+      relationshipsRepository.delete = jest.fn();
+      groupsRepository.delete = jest.fn();
+
+      await controller.delete(req, res, next);
+
+      expect(accountRepository.delete).toHaveBeenCalledTimes(1);
+      expect(profileRepository.delete).toHaveBeenCalledTimes(1);
+      expect(relationshipsRepository.delete).toHaveBeenCalledTimes(1);
+      expect(groupsRepository.delete).toHaveBeenCalledTimes(1);
+      expect(next).toHaveBeenCalledTimes(0);
+      expect(res.status).toHaveBeenCalledTimes(1);
+      expect(res.status).toBeCalledWith(httpStatus.NO_CONTENT);
```

Comment:
httpStatus.OK because it has successfully deleted the document

Reply:
Okay sure I'll change it back to that.